compiler/codegen/src/compile.rs

```diff
@@ -1986,22 +1986,22 @@ impl Compiler {
         Ok(())
     }
 
-    fn compile_dict(&mut self, keys: &[ast::Expr], values: &[ast::Expr]) -> CompileResult<()> {
-        let mut size = 0;
-
-        let (packed_values, unpacked_values) = values.split_at(keys.len());
-        for (key, value) in keys.iter().zip(packed_values) {
-            self.compile_expression(key)?;
-            self.compile_expression(value)?;
-            size += 1;
-        }
-        emit!(self, Instruction::BuildMap { size });
-
-        for value in unpacked_values {
-            self.compile_expression(value)?;
-            emit!(self, Instruction::DictUpdate);
+    fn compile_dict(
+        &mut self,
+        keys: &[Option<ast::Expr>],
+        values: &[ast::Expr],
+    ) -> CompileResult<()> {
+        emit!(self, Instruction::BuildMap { size: 0 });
```

Comment:
This implementation looks like to make performance regression

Reply:
Got it. I'll revert it.